packages/editor/src/store/reducer.js

```diff
@@ -345,16 +76,19 @@ const withSaveReusableBlock = ( reducer ) => ( state, action ) => {
  */
 export const editor = flow( [
 	combineReducers,
+] )( {
+	// Track whether changes exist, resetting at each post save. Relies on
+	// editor initialization firing post reset as an effect.
+	blocks: withChangeDetection( {
+		resetTypes: [ 'SETUP_EDITOR_STATE', 'REQUEST_POST_UPDATE_START' ],
+	} )( ( state = { value: [] }, action ) => {
+		switch ( action.type ) {
+			case 'UPDATE_BLOCKS':
```

Comment:
Why did we rename this from `RESET_BLOCKS`.

We still have a parallel `RESET_POST` and `UPDATE_POST` which respectively replace / merge to post values.

Your new `UPDATE_BLOCKS` effectively inherits the same behavior as what `RESET_POST` is to posts, while `UPDATE_POST` exists and has an entirely different behavior.

Reply:
I understand the reasoning but my reasoning here is that we're updating the blocks of the editor (any update goes through this action) there's no distinction needed between a reset and an update.

I don't feel strongly though.

Reply:
I think we ought to be consistent with the vocabulary, even if the module doesn't necessarily have need for the merge behavior.